### ISSUE TITLE
Accessibility Issue: Grid is keyboard focusable/interactive while overlays are shown

### ISSUE THREAD

**Issue by hoovercj** [ag-grid/ag-grid#5949]
* [x] bug report => see 'Providing a Reproducible Scenario'

Repros at https://www.ag-grid.com/javascript-data-grid/overlays/

**Current behavior**
* Show the loading overlay
* Press "tab" on the keyboard until the focus is on a grid header
* Press enter, causing the sort to change

This is an accessibility issue as screenreader users will be able to access content that they are not intended to interact with, and they will encounter this content **before** they encounter the overlay that describes the loading state of the grid.

This is especially problematic for both screenreader users and sighted keyboard users who will be unable to reasonably access the overlay to invoke any actions it might have such as a "cancel" button.

**Expected behavior**
* Show the loading overlay
* Press "tab" on the keyboard. Focus **never** goes to content that is covered by the overlay and isntead goes directly **to** the overlay.

**Additional Info**
* **AG Grid version:** 28.2.1
* **Browser:** Edge/Chrome
* **User Agent:**  Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36 Edg/108.0.1462.54
* **Language:** All

**Comment by StephenCooper** [comment]
Hi @hoovercj,

Thanks for sharing this with us. I have raised this internally for further investigation.

**Comment by kiril-matev** [comment]
Hello @hoovercj,

Thank you for bringing this up with us. 

The functionality you described wasn't part of the requirements when we built this feature. This is why we have added a feature request in our backlog to track this requirement and implement it in the future.

You can track it with this reference:
**AG-614 [Overlay] Allow overlays to fully cover grid and prevent focusing any grid element**

You can follow the status of feature requests, bugs and releases using our pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

**Now that this requirement is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this feature is added in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/**

Thanks again for bringing this up with us - we appreciate your time!
